### PR TITLE
Omg 302 adjust watcher api

### PR DESCRIPTION
**Pull request by pdobacz**
included:
- `/transaction.xyz` format and scope
- `/account.get_utxos` route and format
- heavy test refactors, reducing integration test brittleness, DRYing
- tweaks to designs, as documented **in slate** (`docs/apis_specs/...`)

excluded:
- `/block.xyz` endpoints (pending design still)
- exits/challenges - pending fixes post MoreVP-compat PR #308 
- demo's won't work - pending fixes post MoreVP-compat PR #308 
- tidying swagger vs slate docs - for now swagger is **outdated**, might be dropped

### CODE REVIEW

**Comment by pnowosie** [comment]
This is not Watcher API - but it'd nice to fix :)
In CC API `/block.get` response there is `number` instead of `blknum`

File: apps/omg_watcher/lib/db/transaction.ex [hunk 1]
```diff
@@ -52,45 +52,35 @@ defmodule OMG.Watcher.DB.Transaction do
...
+      distinct: true,
+      left_join: output in assoc(tx, :outputs),
+      left_join: input in assoc(tx, :inputs),
+      where: output.owner == ^address or input.owner == ^address
+    )
+    |> DB.Repo.all()
+  end
 
-    DB.Repo.all(query)
+  defp base_query_get_last(limit) do
```
Comment (pnowosie) [inline apps/omg_watcher/lib/db/transaction.ex]: This is clever (and it works!)

File: docs/api_specs/includes/informational_api_specs.md [hunk 2]
```diff
@@ -154,7 +157,7 @@ limit | Integer | Maximum number of transactions to return (default 200)
 ## Transaction -  Get All
 
 ```shell
-curl -X POST -H "Content-Type: application/json" http://localhost:4000/transaction.all -d '{"block": "100", "limit": 50}'
+curl -X POST -H "Content-Type: application/json" http://localhost:7434/transaction.all -d '{"block": "100", "limit": 50}'
```
Comment (pnowosie) [inline docs/api_specs/includes/informational_api_specs.md]: Hmm, paramater named `block` is misleading - everywhere we stick with `blknum`

File: apps/omg_watcher/lib/web/controllers/account.ex [hunk 3]
```diff
@@ -30,14 +30,22 @@ defmodule OMG.Watcher.Web.Controller.Account do
   Gets plasma account balance
   """
   def get_balance(conn, params) do
-    # TODO: handle input validation (separate task)
     with {:ok, address} <- Map.fetch(params, "address"),
          {:ok, decoded_address} <- Crypto.decode_address(address) do
-      balance = Account.get_balance(decoded_address)
+      balance = API.Account.get_balance(decoded_address)
```
Comment (pnowosie) [inline apps/omg_watcher/lib/web/controllers/account.ex]: :+1:

File: apps/omg_watcher/lib/web/controllers/transaction.ex [hunk 4]
```diff
@@ -48,12 +46,10 @@ defmodule OMG.Watcher.Web.Controller.Transaction do
   """
   def get_transactions(conn, params) do
     address = get_address(params)
-    limit = Map.get(params, "limit", @default_transactions_limit)
-    {limit, ""} = limit |> Kernel.to_string() |> Integer.parse()
-    # TODO: implement pagination. Defend against fetching huge dataset.
-    limit = min(limit, @default_transactions_limit)
+    limit = get_optional_int("limit", params)
+    blknum = get_optional_int("block", params)
```
Comment (pnowosie) [inline apps/omg_watcher/lib/web/controllers/transaction.ex]:
```suggestion
    blknum = get_optional_int("blknum", params)
```

File: apps/omg_watcher/lib/web/views/utxo.ex [hunk 5]
```diff
@@ -33,21 +33,8 @@ defmodule OMG.Watcher.Web.View.Utxo do
...
-       }) do
-    %{
-      amount: amount,
-      currency: currency,
-      blknum: blknum,
-      txindex: txindex,
-      oindex: oindex,
-      txbytes: tx && tx.txbytes
-    }
+  defp to_view(%DB.TxOutput{} = db_entry) do
```
Comment (pnowosie) [inline apps/omg_watcher/lib/web/views/utxo.ex]: :+1:

File: docs/api_specs/includes/watcher_api_specs.md [hunk 6]
```diff
@@ -48,14 +51,13 @@ Gets all utxos belonging to the given address.
...
 address | Hex encoded string | Address of the account
-limit | Integer | Maximum number of utxos to return (default 200)
 
 
 
 ## Utxo - Get Challenge Data
 
 ```shell
-curl -X POST -H "Content-Type: application/json" http://localhost:4000/utxo.get_challenge_data -d '{"utxo_pos": "10001001"}'
+curl -X POST -H "Content-Type: application/json" http://localhost:7434/utxo.get_challenge_data -d '{"utxo_pos": "10001001"}'
```
[the remark this answers is not in the document]
Comment (pdobacz) [reply docs/api_specs/includes/watcher_api_specs.md]: former: fixed, latter: up to Pawels PR

File: apps/omg_api/lib/api.ex [hunk 7]
```diff
@@ -20,36 +20,33 @@ defmodule OMG.API do
...
     with {:ok, struct_block} <- FreshBlocks.get(hash) do
-      _ = Logger.debug(fn -> " resulted successfully, hash '#{inspect(hash)}'" end)
-      {:ok, Map.from_struct(struct_block)}
-    else
-      error ->
-        _ = Logger.debug(fn -> " resulted with error #{inspect(error)}, hash '#{inspect(hash)}'" end)
-        error
+      {:ok, Block.to_api_format(struct_block)}
     end
+    |> result_with_logging()
```
Comment (pnowosie) [inline apps/omg_api/lib/api.ex]: This potentially will write huge block to the logs (note: previously we just logged the hash of the block)

Comment (pdobacz) [reply apps/omg_api/lib/api.ex]: ah true. This is `debug` though + `inspect` should put ellipsis in its result, so not that bad maybe?

Comment (pnowosie) [reply apps/omg_api/lib/api.ex]: Ah, yes - OK then

**Pull request merged**